### ISSUE TITLE
About the batch size in dataloader

### ISSUE THREAD

**Issue by zx19950617** [j96w/DenseFusion#163]
Hi, guys, I'm confused about how to make the training more efficient.

The batch size in dataloader is always '1', which limits the training speed of the network, right? Now, have you any solution to increase the batch size in the dataloader to speed up the training process? About the buffering idea mentioned in author's reply https://github.com/j96w/DenseFusion/issues/16#issuecomment-475191083, I don't know how to achieve it. Have you updated the code for this idea? I'd like to refer to your solution for study.

Any suggestion will be appreciated a lot! Looking forward to reply!
Best regards

**Comment by MengHao666** [comment]
That is my problem as well

**Comment by j96w** [comment]
Hi, thanks for mention this issue. The reason why the batch size in dataloader is always 1 is that the image crop of each training sample may have different sizes. I have tried the buffering idea. However, it didn't improve the speed very well. Basically, I found that the distribution of the crop size of YCB_Video Dataset is not uniform. Some sizes just have more samples, which makes the training imbalanced. Other methods to improve efficiency are fixed crop size or use padding. However, because of the occlusion in this dataset, especially the synthetic data, there are many cases the 2d bbox of the target object is either very slim or wide which would also waste computation with fixed crop (many pixels belong to other objects or background) or padding (too many zeros). But I think overall the fixed size and padding are still two potential ways to improve efficiency.
